### DISCUSSION TITLE
getting train results after the end of each epoch

### DISCUSSION THREAD

**Discussion by tiago-lam** [ultralytics/ultralytics#4198] [Q&A]
Hello, 

I'm wondering if there is a way to get the training results at the end of each epoch, instead of only getting it at the end of the whole training session.

If someone knows how to do it, please let me know.

Thanks in advance!

**Comment by glenn-jocher** [comment]
@tiago-lam hello!

Absolutely, YOLOv8 provides training results at the end of each epoch. These results include metrics such as precision, recall, and mAP (mean Average Precision), among others. You should see these metrics printed out in your console or terminal after each epoch completes. Additionally, these metrics are logged to a results file that you can review later.

If you're not seeing these, please ensure you're running the latest version of the YOLOv8 repository and check your console output settings. If you continue to face issues, feel free to provide more details, and we'll be happy to help you troubleshoot further. Happy training! 😊🚀
